tests/unittests/client/test_experiment_client.py

```diff
@@ -1228,48 +1270,38 @@ def test_workon_on_error_raise(self):
         def on_error(client, trial, error, worker_broken_trials):
             raise error
 
-        def foo(x):
-            raise NotImplementedError("Do not ignore this!")
-
         with create_experiment(config, base_trial, statuses=[]) as (
             cfg,
             experiment,
             client,
         ):
             with pytest.raises(NotImplementedError) as exc:
-                client.workon(foo, max_trials=5, max_broken=5, on_error=on_error)
+                client.workon(
+                    foo_reraise, max_trials=5, max_broken=5, on_error=on_error
+                )
 
             assert exc.match("Do not ignore this!")
 
     def test_parallel_workers(self, monkeypatch):
         """Test parallel execution with joblib"""
 
-        def foo(x):
-            return [dict(name="result", type="objective", value=x * 2)]
-
-        def optimize(*args, **kwargs):
-            optimize.count += 1
-            return 1
-
         with create_experiment(exp_config=config, trial_config={}, statuses=[]) as (
             cfg,
             experiment,
             client,
         ):
 
-            monkeypatch.setattr(client, "_optimize", optimize)
-            optimize.count = 0
             with client.tmp_executor("joblib", n_workers=5, backend="threading"):
-                client.workon(foo, max_trials=5, n_workers=2)
+                trials = client.workon(foo_1, max_trials=5, n_workers=2)
```

Comment:
max_trials should be adjusted in this test so that we can verify that the runner is indeed running with the correct number of workers. The main goal of this test was to verify this in particular, which was done with `optimize.count`.

Reply:
Is it really part of the Orion testing scope ?
if `multiprocessing.Pool`/`Dask`/`ThreadPoolExecutor` does not spawn the correct number of workers it is not a bug for us to solve

Reply:
No indeed, another option is to test that the backend receives the correct number of workers as argument.

Reply:
I added test_executor_receives_correct_worker_count that does that